ledger/complete/mtrie/node/node.go

```diff
@@ -39,7 +38,7 @@ type Node struct {
 	height    int             // height where the Node is at
 	path      ledger.Path     // the storage path (leaf nodes only)
```

Comment:
I don't think this is a good idea to always include `path` even for non-leaf nodes:
* it consumes unnecessary memory
* it provides missleading return values: if the node is an interior node, it will say "my path is 00..00" even though it is completely different. I think this should be a pointer to `path`, so we can set it to `nil` (same as payload)

Reply:
The path field of nodes will be switched to `*Path` as discussed.

Reply:
Here is a small comparison of the 2 approaches:

Let `n` be the number of key-value elements stored in the trie. The trie contains about `n/2` leafs and `n/2` non-leaf nodes 
- using a `Path` field:  
   - memory usage: `32*n` bytes
   - dynamic pointers:  `0` (only the node pointers need to be tracked by GC)
   - data locality: paths are stored in the same memory block as the node data
   - path values in non-leaf nodes are dummy but that's not an issue as long as we document it IMHO.
   
 - using a `*Path` field:  
   - memory usage: `8*n + 32* (n/2) = 24*n` bytes
   - dynamic pointers:  `n/2` (in addition to the node pointers already tracked by GC)
   - data locality : paths are stored most likely in a different memory block than the node data.
   
   I think the tradeoff is `8*n` bytes of memory Vs extra GC work and data non-locality overhead. 
   I won't start by applying this switch, @AlexHentschel let me know if you think `*Path` still offers a better tradeoff.